File: library/src/main/java/com/james602152002/floatinglabeledittext/validator/RegexValidator.java
package com.james602152002.floatinglabeledittext.validator;

import android.support.annotation.NonNull;

import java.util.regex.Pattern;

/**
 * Created by shiki60215 on 18-1-19.
 */

public class RegexValidator {

    private Pattern pattern;
    private String error_message;
    private boolean matches;

    public RegexValidator(@NonNull String error_message, @NonNull String regex) {
        this.error_message = error_message;
        pattern = Pattern.compile(regex);
    }

    public RegexValidator(@NonNull String error_message, @NonNull String regex, boolean matches) {
        this.error_message = error_message;
        pattern = Pattern.compile(regex);
        this.matches = matches;
    }

    public String getError_message(CharSequence text) {
        if (pattern.matcher(text).matches()) {
            return matches ? error_message : null;
        }
        return error_message;
    }
}

File: library/src/main/java/com/james602152002/floatinglabeledittext/validator/RegexValidator.kt
package com.james602152002.floatinglabeledittext.validator

import java.util.regex.Pattern

/**
 * Created by shiki60215 on 18-1-19.
 */
open class RegexValidator {
    private var pattern: Pattern
    private var error_message: String
    protected var matches = false

    constructor(error_message: String, regex: String) {
        this.error_message = error_message
        pattern = Pattern.compile(regex)
    }

    constructor(error_message: String, regex: String, matches: Boolean) {
        this.error_message = error_message
        pattern = Pattern.compile(regex)
        this.matches = matches
    }

    fun getError_message(text: CharSequence?): String? {
        return if (pattern.matcher(text ?: "").matches()) {
            if (matches) error_message else null
        } else error_message
    }
}